x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/settings/agent_keys/agent_keys.spec.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import { PrivilegeType, ClusterPrivilegeType } from '@kbn/apm-plugin/common/privilege_type';
+import type { RoleCredentials } from '../../../../../services';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../../ftr_provider_context';
+import { expectToReject } from '../../../../../../../apm_api_integration/common/utils/expect_to_reject';
+import type { ApmApiError } from '../../../../../services/apm_api';
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const apmApiClient = getService('apmApi');
+  const samlAuth = getService('samlAuth');
+
+  const agentKeyName = 'test';
+  const allApplicationPrivileges = [PrivilegeType.AGENT_CONFIG, PrivilegeType.EVENT];
+  const clusterPrivileges = [ClusterPrivilegeType.MANAGE_OWN_API_KEY];
+
+  async function createAgentKey(
+    apiClient: (typeof apmApiClient)[keyof typeof apmApiClient],
+    roleAuthc: RoleCredentials,
+    privileges = allApplicationPrivileges
+  ) {
+    return await apiClient({
```

Comment:
could we use `apmApiClient.writeUser` here instead of getting it via parameter?

```suggestion
    return await apmApiClient.writeUser(({
```

Reply:
This suggestion is outdated now. I’ll be using the new `apmApiClient.publicApi` introduced in https://github.com/iblancof/kibana/pull/1.

https://github.com/elastic/kibana/pull/200762/commits/d50e6bfc6751500f04cb09e4421640112c2bd911